components/match2/wikis/clashroyale/opponent_display_custom.lua

```diff
@@ -0,0 +1,110 @@
+---
+-- @Liquipedia
+-- wiki=clashroyale
+-- page=Module:OpponentDisplay/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+-- needed for duo display
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local DisplayUtil = require('Module:DisplayUtil')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+
+-- can not use Module:OpponentLibraries here due to circular requires
+local Opponent = Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+local OpponentDisplay = Lua.import('Module:OpponentDisplay', {requireDevIfEnabled = true})
+local PlayerDisplay = Lua.import('Module:Player/Display', {requireDevIfEnabled = true})
+
+local CustomOpponentDisplay = Table.copy(OpponentDisplay)
+
+CustomOpponentDisplay.BracketOpponentEntry = Class.new(OpponentDisplay.BracketOpponentEntry,
+	function(self, opponent, options)
+		if opponent.type == Opponent.duo then
+			self:createDuo(opponent)
+		end
+
+		self.root = mw.html.create('div'):addClass('brkts-opponent-entry')
+			:node(self.content)
+	end
+)
+
+function CustomOpponentDisplay.BracketOpponentEntry:createDuo(opponent)
+	local playerNode = CustomOpponentDisplay.BlockPlayers({
+		opponent = opponent,
+		overflow = 'ellipsis',
+		showLink = false,
+	})
+	self.content:node(playerNode)
+end
+
+function CustomOpponentDisplay.BracketOpponentEntry:createPlayer(player)
+	local playerNode = PlayerDisplay.BlockPlayer({
+		player = player,
+		overflow = 'ellipsis',
+		showLink = false,
+	})
+	self.content:node(playerNode)
+end
```

Comment:
Why's the override of `create(Solo)Player` needed? Thought CR only needed duos?

Reply:
because they do not want players to link in brackets, i.e. because of line 49: `showLink = false,`

Reply:
We should never have links directly in the brackets?

Reply:
RL wants them and they were the first hence the default for non teams is players link in brackets

open for adjusting it though (and giving RL an overwrite instead)

Reply:
Let's do it. I also asked RL really wants them